src/helpers/indexing-helper.php

```diff
@@ -255,12 +261,18 @@ public function get_unindexed_count() {
 	 *
 	 * @return int The total number of unindexed objects.
 	 */
-	public function get_filtered_unindexed_count() {
+	public function get_filtered_unindexed_count( $limit = false ) {
```

Comment:
We call `get_filtered_unindexed_count` in the background indexing action now instead of `get_unindexed_count`. So what we did now is introduce a new function `get_limited_filtered_unindexed_count` that does the same as `get_filtered_unindexed_count` and also applies the limit.